lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -538,14 +528,22 @@ export class UserAgentApplication {
         throw ClientAuthError.createInvalidInteractionTypeError();
       }
 
-      // prompt user for interaction
-      this.navigateWindow(urlNavigate, popUpWindow);
+      let popUpWindow: Window;
+      if (interactionType === Constants.interactionTypePopup) {
+        // Generate a popup window
+        popUpWindow = this.openWindow(urlNavigate, "_blank", 1, this, resolve, reject);
+        if (!popUpWindow) {
+          // We pass reject in openWindow, we reject there during an error
+          return;
+        }
+      }
+      else {
+        // prompt user for interaction
+        this.navigateWindow(urlNavigate, popUpWindow);
+      }
     }).catch((err) => {
       this.logger.warning("could not resolve endpoints");
       this.authErrorHandler(interactionType, ClientAuthError.createEndpointResolutionError(err.toString), buildResponseStateOnly(request.state), reject);
-      if (popUpWindow) {
-        popUpWindow.close();
```

Comment:
Why was this removed?

Reply:
I assume that was added to close popUp if some of the previous actions fail (access to cache, build url). Right now, show popup is the last action so if popUp won't open it doesn't have to be closed.